include/runtime.h

```diff
@@ -101,6 +102,8 @@ namespace detail {
 		size_t m_num_nodes;
 		node_id m_local_nid;
 
+		std::unique_ptr<divergence_checker_detail::divergence_checker> m_divergence_check;
```

Comment:
```suggestion
		std::unique_ptr<divergence_checker> m_divergence_check;
```
The `divergence_checker` is the public interface, so it should not be in `divergence_checker_detail`!